internal/api/auth/callback.go

```diff
@@ -156,6 +156,14 @@ func (m *Module) CallbackGETHandler(c *gin.Context) {
 		apiutil.TemplateWebPage(c, page)
 		return
 	}
+
+	// Check user permissions on login
+	if !allowedGroup(claims.Groups) {
+		err := gtserror.New("The user is not allowed access to the application")
```

Comment:
I think the error message could be a bit more generous by letting the caller know which groups the user is in; this might make things a bit easier for debugging. For example:

```suggestion
		err := fmt.Errorf("User groups %+v do not include an allowed group", claims.Groups)
```

Or something along those lines. What do you think?

Reply:
I agree.

Reply:
Do you think that the error message should also dump the list of groups the user is currently in?

Reply:
Nah probably not, that's something that could be debugged on the oauth side, besides which it's slightly more sensitive information that might allow someone to get enough info to eg., escalate an attack. Think it's fine to just return what groups GtS considers "allowed".